### PR TITLE
SAMZA-1176: Make TestJoinOperator unit tests safe for concurrent execution

### PR DESCRIPTION
**Pull request by prateekm**

### CODE REVIEW

**Review by vjagadish1989 — APPROVED** [review]
looks good to me, thank you for the contribution. approved

File: samza-core/src/test/java/org/apache/samza/operators/TestJoinOperator.java [hunk 1]
```diff
@@ -46,54 +47,47 @@
...
+
     // push messages to first stream
     numbers.forEach(n -> sot.process(new FirstStreamIME(n, n), messageCollector, taskCoordinator));
     // push messages to second stream with same keys
     numbers.forEach(n -> sot.process(new SecondStreamIME(n, n), messageCollector, taskCoordinator));
 
+
     int outputSum = output.stream().reduce(0, (s, m) -> s + m);
-    assertEquals(outputSum, 110);
+    assertEquals(110, outputSum);
```
Comment (navina) [inline samza-core/src/test/java/org/apache/samza/operators/TestJoinOperator.java]: what were you asserting before :)

Comment (prateekm) [inline samza-core/src/test/java/org/apache/samza/operators/TestJoinOperator.java]: Same thing, I just had the order of expected and actual params reversed leading to confusing messages.

Comment (navina) [inline samza-core/src/test/java/org/apache/samza/operators/TestJoinOperator.java]: Yeah.. It was weird that expected and actual have been swapped. Nvm

**Pull request closed without merging**